tests/test_abstract.py

```diff
@@ -193,16 +195,16 @@ def test_repr():
     f1 = AbstractFunction(P.scalar_mul)
     assert repr(f1) == 'AbstractFunction(scalar_mul)'
 
-    tu1 = AbstractTaggedUnion([[13, s2], [4, l1]])
+    tu1 = AbstractTaggedUnion([[13, s2], [4, to_abstract_test(i16)]])
     assert repr(tu1) == \
-        'AbstractTaggedUnion(U(4 :: [Float[32]], 13 :: Float[32]))'
+        'AbstractTaggedUnion(U(4 :: Int[16], 13 :: Float[32]))'
 
 
 def test_repr_recursive():
     sa = S(t=ty.Int[64])
     ta = T.empty()
-    la = L.empty()
-    la.__init__(ta)
+    la = T.empty()
+    la.__init__([ta])
```

Comment:
I know you can't play with AbstractList for recursive things, but the type you are building here can't match any real value.

Reply:
These types do exist in the pipeline: recursive ADTs become recursive tuples after simplify_types.

Reply:
Yeah, I didn't think of that.